doc/dev/monorepo-bench.md

```diff
@@ -0,0 +1,166 @@
+
+#### One-Shot Benchmarks
+
+These run `dune build <TARGET>` and record the time it takes for the command to
+complete. Under normal operation, there are two one-shot benchmarks.
+
+##### Build From Scratch
+
+This builds the target assuming a clean state. As this
+benchmark can take a long time (> 15 minutes) compared to the expected
+noise in benchmark results, it's run only once.
+
+##### Null Build
+This builds the target assuming it's starting with the target already built.
+This benchmark runs several times to reduce noise. The number of repetitions is
+controlled by the `--num-short-job-repeats` argument.
+
+#### Watch Mode Benchmarks
+
+After running the one-shot benchmarks, the benchmark runner will start Dune in
+watch mode. Once the initial build is complete, the watch mode benchmarks are
+performed. Each watch mode benchmark makes a change to a file in the Monorepo
+(which will trigger a rebuild) and measures how long it takes to rebuild the
+target.
+
+Each file is changed 3 times. The benchmark runner waits for a rebuild after each change:
+ - A benign change is made to a source file.
+ - A compile error is introduced into a source file.
+ - The compile error is fixed.
+
+There are 2 files which are changed in watch mode benchmarks. The intention was
+to change a file in a library with few reverse dependencies and to change a
+file in a library with many reverse dependencies:
+ - The file `src/path.ml` in the library `file_path`, which has few reverse
+   dependencies (within the monorepo)
+ - The file `src/list.ml` in the library `base`, which has many reverse
+   dependencies (within the monorepo)
+
+
+### How to Add New Benchmarks
+
+One-shot benchmarks are defined ad-hoc, so there's no structured way to add more,
+but see the use of the function `measure_one_shot_build` in [this
+file](https://github.com/ocaml-dune/ocaml-monorepo-benchmark/blob/main/dune-monorepo-benchmark-runner/src/main.ml)
+to learn how one might add more one-shot benchmarks.
+
+To add new watch mode benchmarks, edit the file
+[dune-monorepo-benchmark-runner/src/watch_mode_scenarios.ml](https://github.com/ocaml-dune/ocaml-monorepo-benchmark/blob/main/dune-monorepo-benchmark-runner/src/watch_mode_scenarios.ml)
+to add new entries to the list. The fields of list entries are documented
+[here](https://github.com/ocaml-dune/ocaml-monorepo-benchmark/blob/main/dune-monorepo-benchmark-runner/src/watch_mode_file_to_change.mli).
+Each entry specifies a single file to change, and some text within the file to
+change to a something which compiles and something which does not during the
+watch mode benchmark.
+
+To have Dune start running the new benchmark in `current-bench`, do a new release
+of the
+[ocaml-monorepo-benchmark](https://github.com/ocaml-dune/ocaml-monorepo-benchmark)
+by pushing a tag to the repo named the current date like `YYYY-MM-DD.N` where
+`N` is the current day's release number, starting with 0.
+
+Then update `dune`, editing the
```

Comment:
```suggestion
Then update Dune, editing the
```
Is this Dune the buld system or a `dune` file or directory?

Reply:
It refers to the `dune` repository.